File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;
class Solution {
	public int[] solution(int[] array, int[][] commands) {
		int[] answer = new int[commands.length];

		for (int n = 0; n < commands.length; n++) {
			int i = commands[n][0]-1, j = commands[n][1], k = commands[n][2]-1;
			int arr[] = Arrays.copyOfRange(array, i, j);
            Arrays.sort(arr);
			answer[n] = arr[k];

		}
		return answer;
	}
}

File: 프로그래머스/1/42748. K번째수/K번째수.kt
class Solution {
    fun solution(array: IntArray, commands: Array<IntArray>): IntArray {
        var answer = intArrayOf()

        for (i in commands.indices){
            var arr = array.slice(commands[i][0]-1..commands[i][1]-1)
            arr = arr.sorted()
            answer += arr[commands[i][2]-1]

        }

        return answer
    }
}